### DISCUSSION TITLE
Tiny Dataset For Review

### DISCUSSION THREAD

**Discussion by cspayne** [crystalyragui/MARC2RDA#461] [General]
I wanted to provide some examples of where we are at in the transform at this point in time. There's still a LOT of work to be done, including everything we discussed in our meeting this week, but this is what the transform currently outputs when a record is run through it. 

[singlerecord.xml](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/input/singlerecord.xml) is one single BSR record that I transformed.

[singlerecordRDA.rdf](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/singlerecordRDA.rdf) is the direct output of the transform. 

I also post-processed the file to make it more easily human readable, replacing the RDA property numbers with their English lexical aliases in RDF/XML and Turtle format: 
[singlerecordRDA-lexicalaliases.rdf](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/singlerecordRDA-lexicalaliases.rdf)
[singlerecordRDA-lexicalaliases.ttl](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/singlerecordRDA-lexicalaliases.ttl)

You'll notice `fake:marcfield` elements. These are just for testing and review and show which MARC fields have been processed so that the input can be easily compared to the output. 

I also ran a file containing 7 good quality BSR records provided by Crystal to provide a few more examples. 

[2024-07-18.PCCRecords.SmallSet.xml](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/input/2024-07-18.PCCRecords.SmallSet.xml)

[smalldatasetRDA.rdf](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/smalldatasetRDA.rdf)

[smalldatasetRDA-lexicalaliases.rdf](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/smalldatasetRDA-lexicalaliases.rdf)

[smalldatasetRDA-lexicalaliases.ttl](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/outputDataForReview/smalldatasetRDA-lexicalaliases.ttl)

At this stage there are bound to be errors, temporary placeholders, and things that are missing in the code; the transform is far from complete, but hopefully this is helpful in showing what the transform is doing. 

Please feel free to add anything that needs to be discussed here! Or any additional records/types of records that you would like to see transformed.

**Comment by GordonDunsire** [comment]
@cspayne: Looking at the first, single example, there are some issues:

An online resource does not have any items. See LRM 5.3. The access restrictions apply to the manifestation as a whole, so should be transformed as rdamd:P30145 (restriction on access to manifestation).

Remove square brackets from date of publication, place of publication, and name of publisher values. This issue was raised with the previous transform examples.

There is no need for title of manifestation if there is a title proper; title of manifestation is entailed from title proper. This raises the general issue of verbosity in the transform output: should the transform output triples that can be entailed from property hierarchies and inverses?

Fingerprint is not the correct property; it should be identifier for manifestation. Fingerprint is confined to older printed manifestations.

**Reply by cspayne** [reply]
Thank you for reviewing this Gordon, this is really helpful. 
- 506 can be updated to be restriction on access to manifestation for online resources, I will re-open that issue. 
- 264 and 245 were coded before my time on the project, so those issues will need to be re-opened and the mapping sheets and coding updated. I'm hopeful this won't take long for the mappers or the coders, as it seems only small adjustments are needed at this point. 
- The 024 transform to fingerprint was a copy-paste coding error! Thank you for catching that, it's now been corrected.

**Reply by tmqdeborah** [reply]
If anyone would like to see how this Tiny Dataset looks in RIMMF (and already knows how to use it) they can:
- download and install a new version from: https://rimmf.com/w/doku.php?id=rimmf6:start
- Import the attached .nt file of the data into RIMMF (RIMMF requires .nt formatting):
  - Start RIMMF
    - select Tools
    - select Import entity records
    - check External data  [x] at bottom
    - drag and drop attached file onto form
      - read the note about (current) RIMMF transforms:
        - RIMMF uses the canonical set of RDA elements
        - RIMMF assigns RIMMF IRIs for entities (and retains incoming IRIs as Identifiers **if they are provided in the file**)
        - RIMMF (currently) adds a Note on manifestation about the source of the entity description: e.g., “Import of: http://marc2rda.edu/fake/nom/d17e439”
    - assign a folder name and note where the folder is located
    - click on the blue instruction (Click here to run “Change data folder”) to open the file
      - confirm the change
      - click <OK>
- Explore the entities using the Entity Index or switch to the Appellation Index (Tools / Appellation Index)
  - Double-click on any row in either index to see the metadata description set for an entity

**Reply by cspayne** [reply]
> There is no need for title of manifestation if there is a title proper; title of manifestation is entailed from title proper. This raises the general issue of verbosity in the transform output: should the transform output triples that can be entailed from property hierarchies and inverses?

Looking at the code and mapping, title of manifestation includes subfields $n, $p, and $s. Should I change this code to _only_ output title of manifestation if one of these subfields is present? The mapping currently indicates that the title of manifestation should be $n, $p, and $s only when there is no $a at all, is that correct?

**Reply by tmqdeborah** [reply]
@cspayne, I think that what @GordonDunsire is saying is that the 245 title subfields should only be mapped as 'Title proper' instead of both 'Title proper' and 'Title of manifestation'. 

So, what are the subfields that are mapped as 'Title proper'?



If we **were** mapping 'Title of manifestation', then

- subfields $n$p$s appear only when they are appropriate (and $s appears very rarely)
- no, it would not be correct to map $n$p$s without $a (and they should never appear in a MARC record that way).

**Reply by cspayne** [reply]
@GordonDunsire 
Is this what you meant? Should "Title proper" include $n$p$s if present? Or just $a? Do we process $n$p$s at all?

**Reply by GordonDunsire** [reply]
@cspayne: There is no need to map whatever to both title proper and title of manifestation; that is what I said.

The spreadsheet indicates that $a+n+p+s is mapped to title of manifestation, but not to title proper, and that $a on its own is mapped to title proper and not title of manifestation. I vaguely recall the discussion about this, and there were reasons given why anps did not map to title proper. A record with nps but no a is an error, and the best we can do is map it to title of manifestation.

**Reply by GordonDunsire** [reply]
To clarify '$a on its own' means that there is no n, p or s in the field.

**Comment by cspayne** [comment]
I've found the meeting notes from the last time test data was produced for review, and have read through them and watched the meeting recordings (the ones available). The mappings and code were never updated based on these discussions so I would like to go in and do that, as some pretty important decisions were made. 

Is someone who was present during those discussions willing to do a review once I have updated the relevant mappings?

**Reply by GordonDunsire** [reply]
@cspayne: I am happy to review the output, as before.

**Comment by cspayne** [comment]
@GordonDunsire @tmqdeborah @CECSpecialistI @szapoun @SitaKB @corialanus @AdamSchiff @JianPLee @junghaelee @lake44me 

Hello! I have just run the single record through the transform again, and [the output is available here](https://github.com/uwlib-cams/MARC2RDA/tree/main/Working%20Documents/transformationCode/outputDataForReview/20240812) in various formats. 
I wanted to get this out early in the week so that people have a chance to review it and provide feedback. 
I have been working on implementing the mappings for 600, 610, 611, 630, and 650 and that is reflected here. Work on these fields and thorough testing is still in progress, but any feedback is helpful.  
Note that not all feedback from the last output has been implemented at this point, but it is in the works. I am also compiling records with more examples to run, as well as records with 7XX agent and work fields, as I have been coding those as well.

**Reply by GordonDunsire** [reply]
@cspayne: Just a few issues:

The carrier type is online resource, so there are no specific items. Do not mint an item; instead, add the restriction information to rdamd:P30145 'has restriction on access to manifestation'. (Recent decision, I know)

The local part of minted IRIs for subject headings should strip out, in order, the subfield codes, then the spaces and all punctuation except the hyphen. Examples: 'educationhigherunitedstates'; 'englishlanguagestudyandteachingcontinuingeducationspanishspeakers'; 'womenimmigrantseducationhigherunitedstates'. Capitalization can be retained, but using all lower-case is also fine for deduplication.

Conversely, the spaces and all punctuation is retained for the nomen string, and subfield codes are generally replaced by a double hyphen. Examples: 'Education (Higher) -- United States.'; 'English language -- Study and teaching (Continuing education) -- Spanish speakers.'; 'Women immigrants -- Education (Higher) -- United States.'

Otherwise, this is looking good :-)

**Reply by cspayne** [reply]
Thank you! 
I just ran the transform again and output it to the same file(s). 
1. restriction on access has been updated and implemented
2. headings and nomen strings should be looking better
3. 655 has been implemented

**Reply by GordonDunsire** [reply]
@cspayne: I think we can use the compressed heading as the local part of a minted IRI for RDA entities as well as skos concepts. This will deduplicate in the same way as for concepts.

For example, http://marc2rda.edu/fake/place/d19e134, http://marc2rda.edu/fake/place/d19e143, http://marc2rda.edu/fake/place/d19e148, http://marc2rda.edu/fake/place/d19e157, http://marc2rda.edu/fake/place/d19e169, http://marc2rda.edu/fake/place/d19e174, and http://marc2rda.edu/fake/place/d19e179 are better minted as http://marc2rda.edu/fake/place/lcsh/unitedstates.

I think this can be applied to agent, work, place, and timespan entities as well as concepts extracted from subject headings. I hope we can work out something similar for minting nomen IRIs. I see that the nomen local part is the same as the entity local part in the sample single record, but we have to be careful to ensure that this one-to-one association is maintained. That is, see what happens if http://marc2rda.edu/fake/pla/nom/d19e134, etc. are minted as http://marc2rda.edu/fake/pla/nom/lcsh/unitedstates. This will only work if the transform is minting the entity and its aap nomen with the same method; it will not work if we use a subfield 1 or 0 instead of minting the entity. So it would be good to see this applied to the single entry sample.

**Reply by GordonDunsire** [reply]
I have updated the subject headings doc to clarify the local part for place and timespan in the full examples. I will do the same for the other minted entities in due course.

**Reply by cspayne** [reply]
Hi Gordon, I am happy to test this. To clarify, would you like me to also implement using the source and compressed heading in the nomen IRI as well? Or should I only implement this for agent, work, place, and timespan entities?

**Reply by GordonDunsire** [reply]
@cspayne: Yes, try it for nomen, but only if the corresponding agent, work, place, and timespan entities are minted from the same heading. The method does not work for nomens that are related to already existing entity IRIs.

**Reply by cspayne** [reply]
[I did this](https://github.com/uwlib-cams/MARC2RDA/tree/main/Working%20Documents/transformationCode/outputDataForReview/20240812), and the one issue I'm seeing is two nomen strings for `http://marc2rda.edu/fake/pla/nom/lcsh/unitedstates` - both "United States" and "United States."
This is happening because the function that puts together the IRI removes all punctuation, so they end up being the same.

**Reply by GordonDunsire** [reply]
@cspayne: OK, I guess we can't use the deduplication method for nomens. I guess we could retain the punctuation of the nomen string in the local part of the IRI, but then two distinct nomen IRIs would be generated, each mapped as an AAP from LCSH for the minted instance of Place. I think this means that we should only map the entity-nomen relationship as an AP, for all headings and terms (?).

It is still useful to mint the nomens so that the source is recorded, instead of using/inventing another data provenance method.

**We should briefly discuss this at a meeting.**

**Reply by cspayne** [reply]
@GordonDunsire 
Two nomen IRIs is less than one for each occurence of the heading, so this may be a good way to go. Are they not both authorized access points having come from the source, or does the added punctuation make "United States." not an authorized access point?
In the second case I would say yes, we will need to switch to all entity-nomen relationships as an AP because we cannot tell from the string which ones are authorized, even with a source. 
I agree that either way we should mint nomens when a source is provided. 

I don't think there is time in our meeting today to discuss this, but perhaps we can discuss this in the transform meeting tomorrow if you will be attending and bring it to the larger group next week if we need to.

**Reply by GordonDunsire** [reply]
@cspayne: we must make the assumption that there is only one authorized access point per language in an authority source; that is the practical definition of an AAP. SKOS makes the same 'rule'; there can only be one skos:prefLabel per language in a value vocabulary.

We may be able to make a safe assumption about when an end-stop can be removed from a nomen string, under certain conditions. For example if 'United States.' is the preferred controlled name (AAP) of the place, then would it be recorded without the end-stop when it is used as a geographic qualifier? If yes, then we can assume that 'United States' is the AAP, and the end-stop is added as ISBD punctuation; that is, not part of the AAP. I think we need advice from MARC 21 experts about this.

Another option is to examine the source vocabularies and their application guidance, to determine if there is a small set of patterns that can be used by the transform to remove the end-stop or not. In particular, are there specific patterns and guidance for geographic and 'date' topics?

I can't attend the transform meeting (today), but there's no rush to resolve this, I assume, because any decisions can be applied quickly to the transform and test data, and there are other aspects of the test data to be developing while we continue the general discussion. As I indicated during yesterday's discussion, I'd like more time to look at ways reducing the trade-off (and the bigger datasets :-)

**Comment by cspayne** [comment]
I just ran a couple bigger datasets that we've had lying around through the transform so that they can be looked at as well. [All available here](https://github.com/uwlib-cams/MARC2RDA/tree/main/Working%20Documents/transformationCode/outputDataForReview/20240821)

**Reply by GordonDunsire** [reply]
@cspayne: Looking good!

A couple more issues:

In general, the value of a note element should end in a period.

```
<ex:1145038627man> [has note on manifestation] "Copyright notice date: ©2020" .
```

should be:

```
<ex:1145038627man> [has note on manifestation] "Copyright notice date: ©2020." .
```


I wonder if we can improve the mapping of transliterated values. From bigDataset1:

```
"F245 10 $6 880-02 $a Kim Ŏk sijip / $c Kim Ŏk oe chiŭm ; Kim Yong-jik haesŏl."
+ "F880 10 $6 245-02/$1 $a 김 억 시집 / $c 김 억 외 지음 ; 김 용직 해설."
```

=>

```
<ex:1164340179man> [has title proper] "김 억 시집" .
<ex:1164340179man> [has note on manifestation] "Romanized transliteration of title proper: Kim Ŏk sijip." .
<ex:1164340179man> [has statement of responsibility relating to title proper] "김 억 외 지음 ; 김 용직 해설" .
<ex:1164340179man> [has note on manifestation] "Romanized transliteration of statement of responsibility relating to title proper: Kim Ŏk oe chiŭm ; Kim Yong-jik haesŏl." .
```

```
"F264 #1 $6 880-04 $a Kyŏnggi-do P'aju-si : $b Pŏmusa, $c 2020."
+ "F880 #1 $6 264-04/$1 $a 경기도 파주시 : $b 범우사, $c 2020."
```

=> 

```
<ex:1164340179man> [has publication statement] "경기도 파주시 : 범우사, 2020." .
<ex:1164340179man> [has note on manifestation] "Romanized transliteration of publication statement: Kyŏnggi-do P'aju-si : Pŏmusa, 2020." .
<ex:1164340179man> [has name of publisher] "범우사" .
<ex:1164340179man> [has note on manifestation] "Romanized transliteration of name of publisher: Pŏmusa" .
<ex:1164340179man> [has place of publication] "경기도 파주시" .
<ex:1164340179man> [has note on manifestation] "Romanized transliteration of place of publication: Kyŏnggi-do P'aju-si" .
<ex:1164340179man> [has date of publication] "2020" . // from linked field, not field 26X
```

That is, in general:

Transform the linked field as if was the 'main' field.
Transorm the 'main field' to a note field prefixed by boilerplate associated with the label of the 'main' field.
No note is required if the transformed value from the linked field is numeric.

[I am still looking at the end-stop issue ...]

**Reply by cspayne** [reply]
@GordonDunsire This may be something to discuss with the group, as we [previously decided to map linked fields the same as their 'main' counterparts](https://github.com/uwlib-cams/MARC2RDA/wiki/Decisions-Index#iig-6) and updating this would require a not insignificant amount of work in the transform code. 

We should also discuss how we are treating linked fields when we are mapping the 'main' field to a skos:concept. With nomens we are using rdand:P80113 (is equivalent to) "literal value of 880", is there something similar we can do for a concept?
